src/main/java/com/example/tournamentMaker/randomResultGenerator/RandomResultService.java

```diff
@@ -0,0 +1,186 @@
+import com.example.tournamentMaker.tournament.round.CupRoundService;
+import com.example.tournamentMaker.tournament.round.Round;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Service
+@RequiredArgsConstructor
+class RandomResultService {
+    private final TournamentRepository tournamentRepository;
+    private final GameRepository gameRepository;
+    private final FootballStatisticsRepository footballStatisticsRepository;
+    private final ResultService resultService;
+    private final CupRoundService cupRoundService;
+    private final TeamRepository teamRepository;
+    private static final int MIN_YELLOW_CARDS_IN_MATCH = 0;
+    private static final int MAX_YELLOW_CARDS_IN_MATCH = 3;
+    private final Logger logger = LoggerFactory.getLogger(RandomResultService.class);
+
+    public void drawLotRoundsResults(RandomResultRequest request) {
+        Tournament tournament = tournamentRepository.findByName(request.getTournamentName())
+                .orElseThrow(() -> {
+                    throw new NoSuchElementException(Constans.NO_TOURNAMENT_FOUND);
+                });
+        String[] rounds = request.getRoundsToDraw().split("-");
+        int firstRound = Integer.parseInt(rounds[0]);
+        int lastRound = Integer.parseInt(rounds[1]);
+
+        drawLotFootballRounds(firstRound, lastRound, tournament);
+    }
+
+    private void drawLotFootballRounds(int firstRound, int lastRound, Tournament tournament) {
+        IntStream.rangeClosed(firstRound - 1, lastRound - 1).forEach(roundNumber -> {
+            try {
+                Round round = tournament.getRounds().get(roundNumber);
+                drawLotFootballRoundResult(round);
+                if (tournament.getTournamentType() == TournamentType.CUP && !isLastRound(tournament, round)) {
+                    cupRoundService.createNextRoundSchedule(tournament, round);
+                }
+            } catch (ArrayIndexOutOfBoundsException e) {
+                logger.error("There is no round with index " + (roundNumber - 1) +
+                        " check if a tournament schedule has been created");
+            }
+        });
+    }
+
+    private boolean isLastRound(Tournament tournament, Round round) {
+        return tournament.getRounds().size() == round.getTurn();
+    }
+
+    private void drawLotFootballRoundResult(Round round) {
+        round.getGames().forEach(game -> {
+            if (!cupRoundService.isBye(game)) {
+                drawLotFootballGameResult(game);
+            }
+        });
+    }
+    private void drawLotFootballGameResult(Game game) {
+        int hostPoints = RandomUtil.getRandomGoalsNumber();
+        int guestPoints = RandomUtil.getRandomGoalsNumber();
```

Comment:
rozwaz czy warto to robic statyczne. 
Napisales juz do tego testy jednostkowe ? :) Jezeli nie to napisz i daj znac jak poszlo